src/Symfony/Component/HttpKernel/Controller/ContainerControllerResolver.php

```diff
@@ -86,6 +88,16 @@ protected function instantiateController($class)
             return $this->container->get($class);
         }
 
-        return parent::instantiateController($class);
+        $e = null;
+
+        try {
+            return parent::instantiateController($class);
+        } catch (FatalThrowableError $e) {
+        } catch (\ArgumentCountError $e) {
+        }
+
+       if (null !== $e && $this->container instanceof Container && in_array($class, $this->container->getRemovedIds())) {
```

Comment:
Shouldn't this be in the Fwb ControllerResolver instead?

Reply:
@nicolas-grekas WDYT?

Reply:
this would limit the behavior to ppl using the fwb, I don't see any reason to do so